d2/build.gradle

```diff
@@ -19,6 +19,9 @@ dependencies {
   compile externalDependency.jdkTools
   compile externalDependency.zero_allocation_hashing
   compile externalDependency.xchart
+  compile externalDependency.findbugs
```

Comment:
compileOnly?  or does it actually need compile

Reply:
`compileOnly` was the first thing I tried and is technically the right option to use as we don't need findbugs during runtime, but with `compileOnly`, `./gradlew compileTestJava` still fails with the error:

```
> Task :d2-test-api:compileJava FAILED
/Users/abkishor/.gradle/caches/modules-2/files-2.1/org.apache.zookeeper/zookeeper/3.7.1/b598708e153ff74dd2ebb800419df96df8a01c2d/zookeeper-3.7.1.jar(org/apache/zookeeper/server/ZooKeeperServer.class): warning: Cannot find annotation method 'value()' in type 'SuppressFBWarnings': class file for edu.umd.cs.findbugs.annotations.SuppressFBWarnings not found
/Users/abkishor/.gradle/caches/modules-2/files-2.1/org.apache.zookeeper/zookeeper/3.7.1/b598708e153ff74dd2ebb800419df96df8a01c2d/zookeeper-3.7.1.jar(org/apache/zookeeper/server/ZooKeeperServer.class): warning: Cannot find annotation method 'justification()' in type 'SuppressFBWarnings'
/Users/abkishor/.gradle/caches/modules-2/files-2.1/org.apache.zookeeper/zookeeper/3.7.1/b598708e153ff74dd2ebb800419df96df8a01c2d/zookeeper-3.7.1.jar(org/apache/zookeeper/server/ZooKeeperServer.class): warning: Cannot find annotation method 'value()' in type 'SuppressFBWarnings'
/Users/abkishor/.gradle/caches/modules-2/files-2.1/org.apache.zookeeper/zookeeper/3.7.1/b598708e153ff74dd2ebb800419df96df8a01c2d/zookeeper-3.7.1.jar(org/apache/zookeeper/server/ZooKeeperServer.class): warning: Cannot find annotation method 'justification()' in type 'SuppressFBWarnings'
error: warnings found and -Werror specified
1 error
4 warnings
```

The only way I could get rid of it was by using `compile`.

Reply:
ah maybe you need both compileOnly and testCompileOnly.

Reply:
That works, thanks! 

Updated the PR. Also had to add `compileOnly` / `testCompileOnly` to `:d2-int-test` and `:d2-test-api`.

Reply:
Checks are passing as well.